api/src/main/java/jakarta/xml/bind/ContextFinder.java

```diff
@@ -309,11 +313,18 @@ static JAXBContext find(String factoryId,
             return obj.createContext(contextPath, classLoader, properties);
         }
 
-        Class<?> ctxFactory = (Class<?>) ServiceLoaderUtil.lookupUsingOSGiServiceLoader(
-                JAXBContext.JAXB_CONTEXT_FACTORY, logger);
+        @SuppressWarnings({ "rawtypes", "unchecked" })
+        Iterable<Class<?>> ctxFactories = (Iterable) ServiceLoaderUtil.lookupsUsingOSGiServiceLoader(
```

Comment:
`Iterable<Class<? extends JAXBContextFactory>>` would be better; do not see reason to use raw type on the right side when it is already known on the left side...

Reply:
It's a Java lang thing. I don't think it's possible to directly cast. You have to make an indirect cast via either Object or a raw type.

Reply:
just try it. if `ServiceLoaderUtil.lookupsUsingOSGiServiceLoader` returns `<T>` then the type on the left can be `Iterable<Class<? extends JAXBContextFactory>>` and no casting is required on the right side (likely requires JDK11+; there were some improvements in this area)

Reply:
With the <T> introduced in the other method the cast wasn't needed. So resolving this one.